### ISSUE TITLE
Allow opening scenes in text editor

### ISSUE THREAD

**Issue by caryoscelus** [godotengine/godot#6218]
It is sometimes useful to read the scene code or even edit it. But looks like now it's only possible through external means.

To avoid confusion inside editor, it'd be nice to have separation in document list (e.g. groups like "scripts", "scenes", "misc")

**Comment by bojidar-bg** [comment]
Is this (a slightly far-fetched) duplicate of #1390, per chance?

**Comment by caryoscelus** [comment]
I would say it's more of an addition to it, because it means adding some special action of opening scene in text editor (rather than just using file->open) and hopefully some syntax highlighting.

**Comment by akien-mga** [comment]
For now I'll close this as strictly dependent on #1390 (can't happen without the former being implemented), and I'll add a mention there of potential scene-specific features in the text editor. Eventually if/when #1390 is being worked on, this issue could be reopened if need be to track the scene-specific changes separately.
